File: 1535-find-the-winner-of-an-array-game/1535-find-the-winner-of-an-array-game.java
class Solution {
    public int getWinner(int[] arr, int k) {
        int win=0;
        int ele=arr[0];
        for(int i=1;i<arr.length;i++){
            if(arr[i]>ele){
                win=0;
                ele=arr[i];
            }
            if(++win==k){
               break;
            }
        }
        return ele;
    }
}

File: 1535-find-the-winner-of-an-array-game/1535-find-the-winner-of-an-array-game.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
function swap(a,b){
    let temp=a;
    a=b;
    b=temp;
}

var getWinner = function(arr, k) {
    
    
    let final_pos=arr[0];
    let win=0;
    for(let i=1;i <arr.length;i++){
      const v=arr[i];
      if(v<=final_pos)win++;
      else{
        final_pos=v;
        win=1;
      }
      if(win==k)return final_pos;
    }
    return final_pos;
};